content/source/docs/enterprise/saml/identity-provider-configuration-aad.html.md

```diff
@@ -0,0 +1,68 @@
+    - **Sign on URL:** `https://<TFE HOSTNAME>/`
+    ![Screenshot: The "Configure SAML" page of ADD's new app workflow, with the specified settings entered.](./images/sso-aad-saml-configuration.png)
+5. In the User Attributes & Claims section select the pencil and configure the following items:
+    - **Name Identifier value:** `user.mail`
+    ![Screenshot: Modifying the name identifier value.](./images/sso-aad-saml-user-claims-name-identifier.png)
+6. Still in the "User Attributes & Claims" page, configure a user claim to map which team a user belongs to. Under "Manage user claims" page, configure the following:
+    - **Name:** `MemberOf` (This is the default name for TFE's group [attribute][./attributes.html]; the name of this attribute can be changed in [TFE's SAML settings](./configuration.html) if necessary.)
+    - **Source attribute:** (drop-down): user.assignedroles (Custom roles will be created in Azure Active Directory that will be used to map users and groups to TFE teams.)
+    ![Screenshot: Add MemberOf claim](./images/sso-aad-saml-user-claims-memberof.png)
+7. Under the "SAML Signing Certificate" header, download the signing certificate in base64 format
+    ![Screenshot: Download the SAML signing certificate](./images/sso-aad-saml-signing-certificate.png)
+8. Still in the "Single Sign On Configuration" page, under the "Set up <AAD App Name>", copy the following urls that we will enter in our TFE configuration to link TFE to AAD:
+    - **Login URL:**
+    - **Logout URL:**
+    ![Screenshot: Azure Login/Logout URLs](./images/sso-aad-saml-urls.png)
+9. Navigate to `https://<TFE_HOSTNAME>/app/admin/saml` and configure the following:
+    - **Enable SAML single sign-on** (check box): enabled.
+    - **Single Sign-On URL:** Enter the login url from step 8.
+    - **Single Log-out URL:** Enter the logout url from step 8.
+    - **IDP Certificate:** Enter the contents of the PEM (Based64) encoded X.509 certificate captured in step 7.
+
+## Configure Custom Roles for Team Membership Mapping
+
+1. Create teams in TFE as outlined in [TFE Team Membership](./team-membership.html)
+2. Return to the Azure Portal, navigate to "App Registration" page, and search for the application you created for TFE. Select your app and select "Manifest".
+    ![Screenshot: AAD App Registration](./images/sso-aad-saml-app-registration.png)
+3. In the manifest editor, locate the "appRoles" block. This is where you will add additional roles that map users and groups to teams in TFE.
+    ![Screenshot: Manifest Editor](./images/sso-aad-saml-manifest-approles.png)
+4. The "appRoles" block may contain roles automatically generated by AAD. New roles should be added after the system roles and must contain a unique GUID value for the ID value of the new role. You can use a tool such as [GUID Generator](https://www.guidgenerator.com). Click save to add role.
```

Comment:
```suggestion
4. The "appRoles" block may contain roles automatically generated by AAD. New roles should be added after the system roles and must contain a unique GUID value for the ID value of the new role. You can use a tool such as [GUID Generator](https://www.guidgenerator.com) to create these GUIDs. Click "Save" to add the role.
```

Should this be "add the role" or "add the roles"? I wasn't sure.

Reply:
It could be roles if they add multiple roles or it could be role if they just add one.